src/main/java/com/example/training/repository/PolygonRepositoryImpl.java

```diff
@@ -0,0 +1,68 @@
+package com.example.training.repository;
+
+import com.example.training.model.PolygonEntity;
+import lombok.RequiredArgsConstructor;
+import org.jooq.CommonTableExpression;
+import org.jooq.DSLContext;
+import org.jooq.Field;
+import org.jooq.Record1;
+import org.springframework.stereotype.Repository;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import static com.example.training.jooq.tables.Polygon.POLYGON;
+import static com.example.training.repository.utils.PostGisUtils.*;
+import static org.jooq.impl.DSL.*;
+
+@Repository
+@RequiredArgsConstructor
+public class PolygonRepositoryImpl implements PolygonRepository {
+    private final DSLContext dsl;
+
+    @Override
+    public int save(String points) {
+        CommonTableExpression<Record1<String>> geometry_table = name("geometry_table").fields("geometry_column")
+                .as(select(val(ST_MAKE_POLYGON.apply(ST_GEOM_FROM_TEXT.apply(points)))));
+        String geometry = Objects.requireNonNull(select(geometry_table.field("geometry_column"))
+                        .from(geometry_table)
+                        .fetchAny())
+                .into(String.class);
+        return Objects.requireNonNull(dsl.insertInto(POLYGON, POLYGON.SQUARE, POLYGON.GEOMETRY)
+                        .values(ST_AREA.apply(geometry), field(geometry, Object.class))
```

Comment:
you execute select query 2 times. use "with" and insert from select 
https://www.jooq.org/doc/latest/manual/sql-building/sql-statements/insert-statement/insert-select/